api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +344,142 @@ type AzureWebhookReceiverConfig struct {
+	// Kargo to create a complex, hard-to-guess URL, which implicitly serves as a
+	// shared secret.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
+
+	// Actions is a list of actions to be performed when a webhook event is received.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Actions []GenericWebhookAction `json:"actions,omitempty" protobuf:"bytes,2,rep,name=actions"`
+}
+
+// GenericWebhookAction describes an action to be performed on a resource
+// and the conditions under which it should be performed.
+type GenericWebhookAction struct {
+	// Name is the name of the action to be performed.
+	//
+	// +kubebuilder:validation:Enum=Refresh;
+	Name GenericWebhookActionName `json:"action" protobuf:"bytes,1,opt,name=action"`
+
+	// MatchExpression is the validation criteria that must be met for the action to
+	// be performed.
+	//
+	// +optional
+	MatchExpression string `json:"matchExpression,omitempty" protobuf:"bytes,2,opt,name=matchExpression"`
+
+	// Parameters contains additional parameters for the action.
+	//
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+
+	// Targets is a list of selection criteria for the resources on which the
+	// action should be performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes selection criteria for resources to which some
+// action is to be applied.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource.
+	//
+	// +kubebuilder:validation:Enum=Warehouse;
+	Kind GenericWebhookTargetKind `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// Name is the name of the target resource.
+	//
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	// If used with IndexSelector, the results are the combined (logical AND) of the two criteria.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,3,opt,name=labelSelector"`
+
+	// IndexSelector is a selector used to identify cached target resources by cache key.
+	// If used with LabelSelector, the results are the combined (logical AND) of the two criteria.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,4,opt,name=indexSelector"`
```

Comment:
For LabelSelector and IndexSelector, we've documented what the behavior is if they are used in combination with one another, but we haven't documented the behavior if they are used in combination with name.